packages/inter-protocol/src/psm/psm.js

```diff
@@ -179,24 +165,18 @@ export const start = async (zcf, privateArgs, baggage) => {
     const maxAnchor = floorMultiplyBy(afterFee, anchorPerMinted);
     AmountMath.isGTE(maxAnchor, wanted) ||
       assert.fail(X`wanted ${wanted} is more than ${given} minus fees ${fee}`);
-    try {
-      stageTransfer(seat, stage, { In: afterFee }, { Minted: afterFee });
-      stageTransfer(seat, feePool, { In: fee }, { Minted: fee });
-      stageTransfer(
-        anchorPool,
-        seat,
-        { Anchor: maxAnchor },
-        { Out: maxAnchor },
-      );
-      zcf.reallocate(seat, anchorPool, stage, feePool);
-      burnMinted(afterFee);
-    } catch (e) {
-      stage.clear();
-      anchorPool.clear();
-      feePool.clear();
-      // TODO(#6116) someday, reallocate should guarantee that this case cannot happen
-      throw e;
-    }
+    atomicReallocate(
+      zcf,
+      harden([
+        [seat, stage, { In: afterFee }, { Minted: afterFee }],
+        [seat, feePool, { In: fee }, { Minted: fee }],
+        [anchorPool, seat, { Anchor: maxAnchor }, { Out: maxAnchor }],
+      ]),
+    );
+    // TODO The treatment of `burnMinted` here is inconsistent with the
+    // one at the corresponding TODO comment below. This `burnMinted`
+    // doesn't happen if the `atomicReallocate` throws.
```

Comment:
Reviewers, please look at this pair of TODOs and advise. What I tried to do here is preserve the original semantics, even though they seem inconsistent.

Reply:
in giveMinted, fees are extracted from the amounts traded, and we burn fees if the transfer took place. In wantMinted, we unconditionally mint before the transaction, and burn if the transaction fails. The original logic is correct, and you have correctly reproduced it.

Reply:
Thanks. I have revised the comments to explain this. Done.